euphonic/spectra.py

```diff
@@ -447,17 +468,18 @@ class Spectrum1DCollection(collections.abc.Sequence, Spectrum):
     x_tick_labels : Sequence[Tuple[int, str]] or None
         Special tick labels e.g. for high-symmetry points. The int
         refers to the index in x_data the label should be applied to
-    metadata : Dict[str, Any]
-        Contains metadata about the spectrum. Any keys/values are
-        allowed, but values must be JSON serialisable to write to a
-        json file. There are some functional keys:
-          - 'line_data' : List[Dict[str, Any]]. This contains metadata
-                          for each spectrum in the collection, and must
-                          be of length n_entries
+    metadata : Dict[str, Union[int, str]]
```

Comment:
Good point, I think if I indent it suitably  `Dict[str, Union[int, str, Dict[str, Union[int, str]]]` might be ok

Reply:
I realised actually `LineData = Sequence[Dict[str, Union[int, str]]]` so its even more complex, I think in that case using the `LineData` substitution is actually easier

Reply:
I defined LineData as the data for _one_ line, which is why it also works as the type for the "outer" level. Then I messed up and missed out a `Sequence`. Also, it doesn't really work for the outer level because what I wrote means metadata is _either_ a high-level dict _or_ a set of lines - which is wrong. Ugh. How about:

```python
LineData = Sequence[Dict[str, Union[int, str]]]
...
def __init__(...
             metadata: Optional[Dict[str, Union[int, str, LineData]]],
             ...) -> None
```

Reply:
Yep, that's what I ended up going for (I just haven't pushed it yet...)!